[DOCS] Adds semantic search section to kNN search page (#93782)
elastic/elasticsearch: docs/reference/search/search-your-data/knn-search.asciidoc

score = 0.9 * match_score + 0.1 * knn_score
```

The `knn` option can also be used with <<search-aggregations, `aggregations`>>. 
In general, {es} computes aggregations over all documents that match the search. 
So for approximate kNN search, aggregations are calculated on the top `k` 
nearest documents. If the search also includes a `query`, then aggregations are 
calculated on the combined set of `knn` and `query` matches.

[discrete]
[[semantic-search]]
==== Perform semantic search

KNN search enables you to perform semantic search by using a previously deployed 
{ml-docs}/ml-nlp-search-compare.html#ml-nlp-text-embedding[text embedding model]. 

> You can use semantic search to retrieve results based on the intent and the 
> contextual meaning of the search query not only the literal matches of the 
> keywords.

Reviewer: ```suggestion
Instead of literal matching on search terms, semantic search retrieves results
based on the intent and the contextual meaning of a search query.
```

Reviewer: Just a suggestion. The original sentence is a bit long and I found it hard to read.
